core/lib/zksync_core/src/commitment_generator/mod.rs

```diff
@@ -0,0 +1,235 @@
+use std::time::{Duration, Instant};
+
+use anyhow::Context;
+use itertools::Itertools;
+use metrics::{CommitmentStage, METRICS};
+use multivm::zk_evm_latest::ethereum_types::U256;
+use tokio::sync::watch;
+use zksync_commitment_utils::{bootloader_initial_content_commitment, events_queue_commitment};
+use zksync_dal::{ConnectionPool, StorageProcessor};
+use zksync_types::{
+    commitment::{AuxCommitments, CommitmentCommonInput, CommitmentInput, L1BatchCommitment},
+    writes::{InitialStorageWrite, RepeatedStorageWrite, StateDiffRecord},
+    L1BatchNumber, ProtocolVersionId, StorageKey,
+};
+use zksync_utils::h256_to_u256;
+
+mod metrics;
+
+const SLEEP_INTERVAL: Duration = Duration::from_millis(100);
+
+#[derive(Debug)]
+pub struct CommitmentGenerator {
+    connection_pool: ConnectionPool,
+    stop_receiver: watch::Receiver<bool>,
+}
+
+impl CommitmentGenerator {
+    pub fn new(connection_pool: ConnectionPool, stop_receiver: watch::Receiver<bool>) -> Self {
+        Self {
+            connection_pool,
+            stop_receiver,
+        }
+    }
+
+    async fn calculate_aux_commitments(
+        conn: &mut StorageProcessor<'_>,
+        l1_batch_number: L1BatchNumber,
+        protocol_version: ProtocolVersionId,
+    ) -> anyhow::Result<Option<AuxCommitments>> {
```

Comment:
Does this method ever return `Ok(None)`? If not, the return type can be simplified.

Reply:
Good catch, yes, it doesn't return None, I'll remove `Option`